quetz/main.py

```diff
@@ -542,6 +542,24 @@ def post_channel(
     else:
         actions = new_channel.metadata.actions
 
+    includelist = new_channel.metadata.includelist
+    excludelist = new_channel.metadata.excludelist
+
+    if includelist is not None and excludelist is not None:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Cannot use both `includelist` and `excludelist` together.",
+        )
+
+    package_list = []
+    include = False
+    if includelist is not None:
+        package_list = includelist
+        include = True
+    if excludelist is not None:
+        package_list = excludelist
+        include = False
```

Comment:
wouldn't it be better to keep `includelist` and `excludelist`? I think the API would be cleaner that way

Reply:
Hmm, then we would have to pass these 2 variables (`includelist` and `excludelist`). And then, we need to check which one is empty and which one isn't (because one of them will always be empty else we throw `422`). 

So passing these 2 variables where one is kind of just deadweight to the worker (through `**kwargs`) seemed a bit dull to me.

And then, based on which one is not empty, we would need to call `check_package_membership` on the one which is not empty. This kind of thing would again lead us to the `if` statement such as:

`if ((includelist!=[] and check_package_membership(package_name, includelist)) or (excludelist!=[] and (not check_package_membership(package_name, excludelist)))`

This is even more complicated than what we have IMO.

Plus, there is this issue of default values. With our current implementation, we can supply `include` as `False` and `package_list` as `[]` which is essentially equivalent to `excludelist=[]` or saying that `exclude nothing` aka `include everything`.

But with 2 separate lists, what could be the default value for `includelist`? We wouldn't want to fetch the data to calculate this.

Reply:
In Python, an empty list is "falsy" so `includelist and check_package_membership(name, includelist)` isn't so bad.

Also one could move the "empty" check into the `check_package_membership` function, then it would be only one function call.

The condition would be:

`if check_package_membership(package_name, include_list) or not check_package_membership(package_name, exclude_list):` which looks like a huge improvement in terms of readability to me :)

Reply:
This condition unfortunately doesn't work. I have implemented it a bit differently. Hope that works.